orderer/consensus/follower/follower_chain.go

```diff
@@ -7,41 +7,133 @@ SPDX-License-Identifier: Apache-2.0
+// from the consenters set.
 //
 // The follower is in status "onboarding" when it pulls blocks below the join-block number, or "active" when it
 // pulls blocks equal or above the join-block number.
+//
+// The follower return clusterRelation "member" when the join-block indicates the orderer is in the consenters set,
+// i.e. the follower is performing on-boarding for an etcdraft.Chain. Otherwise, the follower return clusterRelation
+// "follower".
 type Chain struct {
-	Err error
-	//TODO skeleton
+	mutex    sync.Mutex    // Protects the start/stop flags & channels, all the rest are immutable or accessed only by the go-routine.
+	started  bool          // Start once.
+	stopped  bool          // Stop once.
+	stopChan chan struct{} // A 'closer' signals the go-routine to stop by closing this channel.
+	doneChan chan struct{} // The go-routine signals the 'closer' that it is done by closing this channel.
+
+	support Support
+	options Options
+	logger  *flogging.FabricLogger
+
+	joinBlock  *common.Block // The join-block the follower was started with.
+	lastConfig *common.Block // The last config block from the ledger. Accessed only by the go-routine.
+
+	createBlockPullerFunc CreateBlockPuller // A method that creates a block puller on demand
+	chainCreationCallback CreateChain       // A method that creates a new consensus.Chain for this channel, to replace the current follower.Chain
+
+	cryptoProvider bccsp.BCCSP
+	amIInChannel   cluster.SelfMembershipPredicate
+}
+
+// NewChain constructs a follower.Chain object.
+func NewChain(
+	support Support,
+	joinBlock *common.Block,
+	options Options,
+	createBlockPullerFunc CreateBlockPuller, // A method that creates a block puller on demand
+	chainCreationCallback func(),
+	cryptoProvider bccsp.BCCSP,
+	amIInChannel cluster.SelfMembershipPredicate,
+) (*Chain, error) {
+	chain := &Chain{
+		stopChan:              make(chan (struct{})),
+		doneChan:              make(chan (struct{})),
+		support:               support,
+		joinBlock:             joinBlock,
+		options:               options,
+		logger:                options.Logger.With("channel", support.ChannelID()),
+		createBlockPullerFunc: createBlockPullerFunc,
+		chainCreationCallback: chainCreationCallback,
+		cryptoProvider:        cryptoProvider,
+		amIInChannel:          amIInChannel,
+	}
+
+	if joinBlock == nil {
+		chain.logger.Infof("Created with a nil join-block, ledger height: %d", support.Height())
+	} else {
+		if joinBlock.Header == nil {
+			return nil, errors.New("block header is nil")
+		}
+		if joinBlock.Data == nil {
+			return nil, errors.New("block data is nil")
+		}
+
+		chain.logger.Infof("Created with join-block number: %d, ledger height: %d", joinBlock.Header.Number, support.Height())
+	}
+
+	return chain, nil
 }
 
 func (c *Chain) Order(_ *common.Envelope, _ uint64) error {
-	return c.Err
+	return errors.Errorf("orderer is a follower of channel %s", c.support.ChannelID())
```

Comment:
unrelated to this PR: It can be nice if the follower can send the transaction to one of the consenters via the endpoint in the consenter metadata, and it will forward onwards the transaction. 

Then, we get some sort of opaqueness to the user as it doesn't care who is a follower and who is not.

Reply:
That is an interesting idea. However for now I think the main reference scenario is that we create a follower for a channel even before we admit it to the channel endpoints and consenters set, and only after on-boarding a config update admits it into both endpoints and consenters set.